velox/dwio/text/writer/TextWriter.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/common/compression/Compression.h"
+#include "velox/common/config/Config.h"
+#include "velox/dwio/common/DataBuffer.h"
+#include "velox/dwio/common/FileSink.h"
+#include "velox/dwio/common/FlushPolicy.h"
+#include "velox/dwio/common/Options.h"
+#include "velox/dwio/common/Writer.h"
+#include "velox/dwio/common/WriterFactory.h"
+#include "velox/dwio/text/writer/BufferedWriterSink.h"
+#include "velox/vector/ComplexVector.h"
+
+namespace facebook::velox::text {
+
+struct WriterOptions : public dwio::common::WriterOptions {
+  int64_t defaultFlushCount = 10 << 10;
+};
+
+/// Encodes Velox vectors in TextFormat and writes into a FileSink.
+class TextWriter : public dwio::common::Writer {
+ public:
+  const static char SOH = '\x01';
```

Comment:
Shall we put this in a separate file TextFileTraits.h? 
```
class TextFileTraits {
  const static char SOH = '\x01';
   ...
}
```
Could you also comment what are these delimiters used for? Thanks!